packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -298,47 +317,45 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
         external
         whenNotPaused
     {
-        // Prevent users from creating a deposit transaction where this address is the message
-        // sender on L2. Because this is checked here, we do not need to check again in
-        // `finalizeWithdrawalTransaction`.
-        if (_tx.target == address(this)) revert BadTarget();
+        // Make sure that the target address is safe.
+        if (_isUnsafeTarget(_tx.target)) {
+            revert OptimismPortal_BadTarget();
+        }
 
         // Fetch the dispute game proxy from the `DisputeGameFactory` contract.
-        (GameType gameType,, IDisputeGame gameProxy) = disputeGameFactory.gameAtIndex(_disputeGameIndex);
-        Claim outputRoot = gameProxy.rootClaim();
-
-        // The game type of the dispute game must be the respected game type.
-        if (gameType.raw() != respectedGameType.raw()) revert InvalidGameType();
-
-        // The game type of the DisputeGame must have been the respected game type at creation.
-        // eip150-safe
-        try gameProxy.wasRespectedGameTypeWhenCreated() returns (bool wasRespected_) {
-            if (!wasRespected_) revert InvalidGameType();
-        } catch {
-            revert LegacyGame();
+        (,, IDisputeGame gameProxy) = disputeGameFactory.gameAtIndex(_disputeGameIndex);
```

Comment:
Can we remove all DGF references from the portal by having this method take the `gameProxy` as an input and we call `anchorStateRegistry.isGameRegistered(gameProxy)` here instead? I'm guessing the problem is this break the interface for existing users but it would a nice architectural cleanup

Reply:
Yeah it just super breaks the interface :-/ main reason it's still in there is because of all the old L2OutputOracle chains that took an index as a parameter

Reply:
We can get rid of this once we're fully off of L2OutputOracle

Reply:
I'd say the right time to get rid of this is the upgrade after interop when we also get rid of the legacy `proveWithdrawalTransaction` function for output roots. We could even have the super roots proving function PR take a dispute game as the input which would get us 50% of the way there.